redaxo/src/core/lib/console/setup/run.php

```diff
@@ -0,0 +1,424 @@
+
+                rex::setProperty('db', $config['db']);
+                try {
+                    $err = rex_setup::checkDb($config, $redaxo_db_create);
+                } catch (PDOException $e) {
+                    $err = 'The following error occured: ' . $e->getMessage();
+                }
+
+                if ('' !== $err) {
+                    $io->error($err);
+                }
+            } while ('' !== $err);
+        } else {
+            $config['db'][1]['host'] = $input->getOption('db-host');
+            $config['db'][1]['login'] = $input->getOption('db-login');
+            $config['db'][1]['password'] = $input->getOption('db-password');
+            $config['db'][1]['name'] = $input->getOption('db-name');
+
+            $redaxo_db_create = true === $input->getOption('db-createdb');
+
+            rex::setProperty('db', $config['db']);
+            try {
+                $err = rex_setup::checkDb($config, $redaxo_db_create);
+            } catch (PDOException $e) {
+                $err = 'The following error occured: ' . $e->getMessage();
+            }
+
+            if ('' !== $err) {
+                $io->error($err);
+                return 1;
+            }
+        }
+        $io->success('Database connection successfully established');
+
+        // ---------------------------------- step 5 . create db / demo
+        $io->title('Step 5 of 6 / Database');
+
+        // Search for exports
+        $backups = [];
+        foreach (rex_backup::getBackupFiles('') as $file) {
+            if ('.sql' != substr($file, strlen($file) - 4)) {
+                continue;
+            }
+            $backups[] = substr($file, 0, -4);
+        }
+
+        $createdbOptions = [
+            'normal' => 'Setup database',
+            'override' => 'Setup database and overwrite it if it exitsts already (Caution - All existing data will be deleted!',
+            'existing' => 'Database already exists (Continue without database import)',
+            'update' => 'Update database (Update from previous version)',
+        ];
+        if (count($backups) > 0) {
+            $createdbOptions['import'] = 'Import existing database export';
+        }
+
+        if (!$input->getOption('db-setup')) {
+            $createdb = $io->askQuestion(new ChoiceQuestion('Setup database', $createdbOptions, 'normal'));
+        } else {
+            $validOptions = array_keys($createdbOptions);
+            $createdb = $input->getOption('db-setup');
+            if (!in_array($createdb, $validOptions, true)) {
+                throw new InvalidArgumentException('Unknown db-setup value "'.$createdb.'". Valid values are ' . implode(', ', $validOptions));
+            }
+        }
+
+        $tables_complete = ('' == rex_setup_importer::verifyDbSchema()) ? true : false;
+
+        if ('update' == $createdb) {
+            $useUtf8mb4 = $this->getDbCharset();
+            $config['utf8mb4'] = 'utf8mb4' === $useUtf8mb4;
```

Comment:
Das ist keine Option in der config.yml, sondern wird in die DB in rex_config geschrieben (`rex::setConfig`).